File: php/features.php
<?php
declare(strict_types=1);

// InnoCyberAuthentication SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class InnoCyberAuthenticationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new InnoCyberAuthenticationBaseFeature();
            case "test":
                return new InnoCyberAuthenticationTestFeature();
            default:
                return new InnoCyberAuthenticationBaseFeature();
        }
    }
}

File: py/features.py
# InnoCyberAuthentication SDK feature factory

from feature.base_feature import InnoCyberAuthenticationBaseFeature
from feature.test_feature import InnoCyberAuthenticationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: InnoCyberAuthenticationBaseFeature(),
        "test": lambda: InnoCyberAuthenticationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
